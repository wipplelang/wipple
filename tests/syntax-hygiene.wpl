get : syntax {
  get 'x -> {
    x : "Inner"
    'x
  }
}

x : "Outer"
show (get x)

let : syntax {
  let ('x : 'v) in 'y -> {
    'x : 'v
    'y
  }
}

let (foo : 42) in (show foo)
