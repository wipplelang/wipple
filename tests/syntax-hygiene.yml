code: |
    get : syntax {
        get 'x -> {
          x : "Inner"
          'x
        }
    }

    x : "Outer"
    show (get x)

    let : syntax {
        let ('x : 'v) in 'y -> {
          'x : 'v
          'y
        }
    }

    let (foo : 42) in (show foo)

output: |
    Outer
    42

diagnostics: |
    warning: unused variable
      ┌─ test:9:7
      │
    3 │       x : "Inner"
      │       - actual warning occurred here
      ·
    9 │ show (get x)
      │       ^^^ `x` is never used
      │
      = for more information, see https://wipple.dev/playground/?lesson=errors/unused-variable
