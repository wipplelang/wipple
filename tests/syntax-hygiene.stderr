warning: 
  ┌─ test:9:7
  │
3 │     x : "Inner"
  │     - actual warning occurred here
  ·
9 │ show (get x)
  │       ^^^ `x` is never used
  │
  = for more information, see https://wipple.dev/playground/?lesson=errors/unused-variable

